Remove numbers from documentation filenames in Fleet repo

eashaw opened the pull request:
Closes #3274

Changes:
- Removed numbers from filenames and folder names in the `docs` folder
- Added a new meta tag to the documentation pages: `pageOrderInSection` and set values based on the current documentation order e.g., `<meta name="pageOrderInSection" value="200">`. The value can be 
- Changed the documentation sort function to sort pages by the pageOrderInSection value.
- Updated all links in the Fleet Repo to documentation (Excluding links to a specific commit on GitHub) to use new file names
- Updated `build-static-content.js` to use the new meta value and throw an error if it is invalid.
- Added a handbook entry about ordering doc pages and `pageOrderInSection` meta tags.


# Checklist for submitter

If some of the following don't apply, delete the relevant line.

- [x] Manual QA for all new/changed functionality

mikermcneil commented on handbook/community.md:
```diff
@@ -43,12 +43,12 @@ Keep headings brief and organize them in a logical order:
 
 ### Adding a link to the Fleet docs
-You can link documentation pages to each other using relative paths. For example, in `docs/01-Using-Fleet/01-Fleet-UI.md`, you can link to `docs/01-Using-Fleet/09-Permissions.md` by writing `[permissions](./09-Permissions.md)`. This will be automatically transformed into the appropriate URL for `fleetdm.com/docs`.
+You can link documentation pages to each other using relative paths. For example, in `docs/Using-Fleet/Fleet-UI.md`, you can link to `docs/Using-Fleet/Permissions.md` by writing `[permissions](./Permissions.md)`. This will be automatically transformed into the appropriate URL for `fleetdm.com/docs`.
```
@eashaw Can we link to fleetdm.com here instead?  (ie. for links to docs from within the handbook?)

The auto-rewriting of doc links doesn't work properly for handbook, IIRC

mikermcneil replied:
Also I'm pretty sure the Permissions.md link won't actually work regardless (check the path)

eashaw replied:
Those two aren't actually links, they're [examples in the handbook](https://fleetdm.com/handbook/community#adding-a-link-to-the-fleet-docs).

mikermcneil replied:
aHHHHHHHHHH sorry.  Fooled me!

mikermcneil commented on website/scripts/build-static-content.js:
```diff
@@ -312,9 +312,32 @@ module.exports = {
+                  throw new Error(`Failed compiling markdown content: A Non FAQ or README Documentation page is missing a pageOrderInSection meta tag (<meta name="pageOrderInSection" value="">) at "${path.join(topLvlRepoPath, pageSourcePath)}".  To resolve, add a meta tag with a number higher than 0.`);
+                }
+              }
+
```
this part LGTM

mikermcneil commented on cmd/fleetctl/preview.go:
```diff
@@ -30,7 +30,7 @@ import (
 const (
 	downloadUrl             = "https://github.com/fleetdm/osquery-in-a-box/archive/%s.zip"
-	standardQueryLibraryUrl = "https://raw.githubusercontent.com/fleetdm/fleet/main/docs/01-Using-Fleet/standard-query-library/standard-query-library.yml"
+	standardQueryLibraryUrl = "https://raw.githubusercontent.com/fleetdm/fleet/main/docs/Using-Fleet/standard-query-library/standard-query-library.yml"
```
What about people on an older version of `fleetctl`?  What happens when we make this change to GitHub?  Will those folks stop being able to use `fleetctl preview`?  As long as there is either:
1. a good error message that fleetctl shows asking the user to update fleetctl (AND if the new version of fleetctl is actually published as soon as we merge this)
2. or `fleetctl preview` just fails gracefully and doesn't include the standard queries (AND the new version of fleetctl is actually published)

Then I'm cool with this.

But I think we might need to think through a different solution.  e.g. We could leave the file in the old location for a while.  @zwass up to you

zwass commented:
Maybe those changelog links that are failing the check should be links to `fleetdm.com/docs` rather than in the repo?

> We could leave the file in the old location for a while.

I checked and `preview` will bail out if it can't find the file. I think we should leave it (or just a copy?) in the original location for now.

eashaw commented:
I updated this PR with these changes:
- Moved the standard query library in its old location and updated references/links to it (`/docs/01-using-fleet/standard-query-library/`)
- Updated non-commit specific doc links in the changelog to go to fleetdm.com/docs

zwass reviewed (approved):
I didn't test that the website builds and the links work properly, but the reorganization lgtm.

mike-j-thomas reviewed (changes requested):
Looks good. Just a couple of fixes.
